cibuildwheel/linux.py

```diff
@@ -115,25 +129,72 @@ def build(project_dir, package_name, output_dir, test_command, test_requires, be
             gid=os.getgid(),
         )
 
-        docker_process = subprocess.Popen([
+        container_name = 'cibuildwheel-{}'.format(uuid.uuid4())
+
+        command = [
+            'docker',
+            'run',
+            '--env',
+            'CIBUILDWHEEL',
+            '--name', container_name,
+            '-i',
+            '-v', '/:/host', # ignored on Circle
+            docker_image,
+            '/bin/bash',
+        ]
+        print('docker command: {}'.format(command))
+        docker_process = subprocess.Popen(
+            command,
+            stdin=subprocess.PIPE,
+            universal_newlines=True,
+        )
+
+        # It will take a bit for docker to get the container up and
+        # running.  Keep trying to copy in the project directory until
+        # it succeeds.  There is a timeout to avoid spinning forever.
+
+        timeout = 180
+        start = monotonic.monotonic()
```

Comment:
Do we really need this `monotonic` dependency? I can see it's maybe good practice to not trust your clock, but if it's just to timeout Docker, wouldn't `time` just do? Especially since `cibuildwheel`is installed again and again on the CI services, it seems nice to limit the amount of dependencies.

Otherwise, if necessary to use `monotonic`, let's document why, so no one will be asking that same question again.

Reply:
It's proper to avoid affects of system time getting changed. It's unlikely to matter in practice. I'm surprised the added dependency got accepted.  :]  If dropping the dependency I would use monotonic in py3 and time in py2 in suppose.  We could probably vendor it as well... But I expect we don't get there.

Reply:
Myeah, but we're mainly running this for a short time on a protected image (where no one's playing with the time), anyway, on the CI servers, so I don't know if this precaution is worth it?

If time does change:
- Either the build fails immediately, if time moved forward and docker wasn't ready. Restarting the job will work.
- Or the build itself will stall for a bit, until a longer timeout is reached (or until the main build times out).

I guess this is up to @joerick? But I'd vote to get rid of it. I've seen Travis CI fail more times because of random errors (apt install failing, or simply no output, or ..) than because of `cibuilwheel`, so as long as nothing bad, undetected happens (such as corrupt wheels), I'm not sure about the dependency.

Reply:
(So even using `time` overall would be fine with me, to keep the Python code equal for 2 and 3?)

Reply:
monotonic:

![download](https://user-images.githubusercontent.com/1244307/45256021-75518100-b388-11e8-9d9b-eb2f33e6b5ab.jpg)

I'm okay with the dependency. 

Also the `monotonic` code is sufficiently hairy that I'd rather not have to deal with it :P

Reply:
@YannickJadoul yeah, I'm not strongly attached to monotonic in this case.  Though it generally is a pet peeve of mine.  I won't be particularly bothered if you talk @joerick out of it.  Mostly I just started with what I considered best practice and then @joerick went and accepted it without giving you a chance to review.  :]

Reply:
Eh, it's still @joerick's call. I don't really mind, but I thought it might be worth explicitly making the decision.